### ISSUE TITLE
Size of the Agda git repo (dev only)

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#5242]
Since recently, I noticed that `git pull` usually pulls in around 8MB.  (Of course, not consecutive pulls, but whenever I return to Agda development the next day or a couple of days later.)
I am not sure what causes this, maybe the nightly build `gh-pages` branch?

Anyhow, I observe that the repo is quite big now (2021-02-26):
```
$ du -d0 -h .git
589M	.git
```
(`git gc` doesn't reduce this by more than 1%).

I have a version from last November (2021-11-23) that is only _one quarter_ in size:
```
$ du -d0 -h ~/tmp/agda/.git
129M	/Users/abel/tmp/agda/.git
```

Even a fresh clone (today, 2021-02-26) is around 300MB:
```
$ du -d0 -h .git
295M	.git
```

Anyone making similar observations?

Btw, I am not sure I can "unsubscribe" locally from the `gh-pages` branch.  Following https://stackoverflow.com/questions/3046436/how-do-you-stop-tracking-a-remote-branch-in-git I deleted the remote tracking branch via
```
git branch -d -r origin/gh-pages
```
but I need to wait a couple of days to see whether it has an effect on the `pull` volume.

**Comment by jespercockx** [comment]
> I have a version from last November (2021-11-23) that is only one quarter in size

You have a version of Agda from 9 months in the future? Maybe you could copy some code from it, would save us a lot of work for the release ;)

But you're right that the repository does seem much slower to clone than it used to.

**Comment by gallais** [comment]
I've installed [git-sizer](https://github.com/github/git-sizer/) and it points at 8a02d2470d553f70042a8a99fa3641e724e9635e being
the biggest commit (100M). This is indeed a haddock commit.

The `docs/` directory on `gh-pages` is a whooping 150M when all of Agda's `src/`
is just 7M in comparison. The biggest file in that `docs/` directory is almost as
big as the entirety of Agda's sources at 5.5M of html: `Agda-Compiler-Backend.html`.
That being said it's all text with a lot of repetitions so it should compress
really well (in a gzipped tarball it's down to 293K for instance) and thus download
fairly quickly.

I'm not sure what's going on there.

**Comment by ice1000** [comment]
Suggest `git gc --aggressive --prune` in favor of plain `git gc`

**Comment by andreasabel** [comment]
> I've installed [git-sizer](https://github.com/github/git-sizer/)

I considered this, but don't have enough space to install a new PL.

**Comment by gallais** [comment]
> I considered this, but don't have enough space to install a new PL.

It was available on the debian repos, I did not need to compile it myself.

**Comment by ice1000** [comment]
You could also download binary releases https://github.com/github/git-sizer/releases/tag/v1.3.0

I think we could learn from this. Providing binary releases is extremely important for a software. Agda used to force me to install by compiling it from source, and that's one of the many reasons why I became an Agda contributor eventually. <details><summary>... (trigger warning!)</summary>
I feel like I was raped and fell in love with the offender (weird analogy I have to admit).
</details>

If Agda could be installed as convenient as git-sizer, I would love Agda and be a passer-by of Agda. Now I hate Agda (I had to help many people installing it. GHC is a shit-mountain, and Agda is written in this language), but I became one of you people here.

I'm sorry, I talk too much

**Comment by andreasabel** [comment]
@ice1000:  Life paths are strange.  I only became a type theorist because the class on TT I attended left me utterly confused...

**Comment by gallais** [comment]
@ice1000 I'd avoid this kind of unnecessarily graphic imagery: you don't know
what other contributors' lives have been like and this could ruin their day by
invoking horrible memories to make an unrelated point.

**Comment by andreasabel** [comment]
@gallais @MatthewDaggitt 
I wouldn't like to censor this kind of metaphorical expression (I am fond of metaphors even of a such drastic kind).  But I acknowledge the possibility of trigger.  

I thought about what technical solution there could be, and there is one: text can be folded/unfolded in comment fields.

@ice1000 
I took the liberty to place your metaphor under a trigger warning, to accommodate both sensitivities and freedom of expression.

**Comment by ice1000** [comment]
> @ice1000 I'd avoid this kind of unnecessarily graphic imagery: you don't know
> what other contributors' lives have been like and this could ruin their day by
> invoking horrible memories to make an unrelated point.

Yes, I am very sorry for that. I think I might violate some code of conduct, and I wish to delete the post. Unfortunately it'll be available in the edit history anyway, so I'll keep AA's modification. Well, maybe I'll mark it as off-topic.

**Comment by ice1000** [comment]
I think I don't know how to mark a comment as off-topic, sorry...

**Comment by L-TChen** [comment]
One possibility is to publish them to another repo and delete the `gh-pages` branch completely. Does it sound like a good idea?

**Comment by ice1000** [comment]
Another solution will be make the `gh-pages` branch a single-commit branch, and let the CI amend + force push to it

**Comment by andreasabel** [comment]
2026-02-28: Now it is 2 GB.
> du -d0 -h .git
> 2.1G	.git

```
$ git-sizer
```
Processing blobs: 136262                        
Processing trees: 187897                        
Processing commits: 58137                        
Matching commits to trees: 58137                        
Processing annotated tags: 103                        
Processing references: 7316                        
| Name                         | Value     | Level of concern               |
| ---------------------------- | --------- | ------------------------------ |
| Biggest objects              |           |                                |
| * Trees                      |           |                                |
|   * Maximum entries      [1] |  3.65 k   | ***                            |
| * Blobs                      |           |                                |
|   * Maximum size         [2] |  13.9 MiB | *                              |

[1]  7d566e984189ced8df6dddf51e87a666f41aabd5 (refs/heads/issue-4150:test/Fail)
[2]  f51fa67529b96ee83ae75db60e190ac738b1e625 (refs/remotes/NathanielB123/gh-pages:docs/src/Agda.Syntax.Parser.Parser.html)

I guess I have too many branches in my working repo.
My bisect repo is much smaller:
> du -hd0 .git
> 232M	.git

**Comment by andreasabel** [comment]
Fixed by @L-TChen 's cb1ea6808e5a458b16a50772ab8b17435ec04c19, I suppose.
